### PR TITLE
ft(google-auth): allow a user to login using google [187364805]

### PR DESCRIPTION
**Pull request by Aldot-02**
- A user should be able to login using google
- Added the google auth button in the components (with full authentication functionality) as it will be utilized both on the login and signup page
- The component is being imported into the dummy signup page now for testing purpose, because the entire signup designs are assigned to another peer
- Added the tests for this functionality in the tests folder

### CODE REVIEW

**Review by tuyishimehope — APPROVED** [review]
LGTM

**Review by martinemahirwe — APPROVED** [review]
LGTM

**Review by tuyishimehope — APPROVED** [review]
LGTM

**Review by martinemahirwe — APPROVED** [review]
LGTM

File: src/components/authcomponents/GoogleBtn.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import image2 from '../../assets/Google.png';
+
+const GoogleBtn = () => {
+  const handleGoogleLogin = () => {
+    const rootUrl = 'https://accounts.google.com/o/oauth2/v2/auth';
+
+    const options = {
+      redirect_uri:
+        'https://staging-e-commerce-serge-30-bn.onrender.com/api/v1/auth/google/callback',
```
Comment (SergeInGithub) [inline src/components/authcomponents/GoogleBtn.tsx]: Let's not expose these URLs... use .env

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Nice work so far... In addition to the reviews below, Improve all components and add another prop to them for classnames... such that one can modify the preconfigured styles of the component with their own. @Aldot-02

**Comment by Aldot-02** [comment]
> Nice work so far... In addition to the reviews below, Improve all components and add another prop to them for classnames... such that one can modify the preconfigured styles of the component with their own. @Aldot-02

@SergeInGithub, I adjusted this for the google button component as the other components are dummy, but I communicated to @martinemahirwe who is working on signup using real components to consider this suggestion.

**Review by tuyishimehope — APPROVED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**